postfix/init.sls

```diff
@@ -17,33 +17,21 @@ postfix:
 
 # manage /etc/aliases if data found in pillar
 {% if 'aliases' in pillar.get('postfix', '') %}
-  {%- set need_newaliases = False %}
-  {%- set file_path = postfix.aliases_file %}
-  {%- if ':' in file_path %}
-    {%- set file_type, file_path = postfix.aliases_file.split(':') %}
-  {%- else %}
-    {%- set file_type = default_database_type %}
-  {%- endif %}
-  {%- if file_type in ("btree", "cdb", "dbm", "hash", "sdbm") %}
-    {%- set need_newaliases = True %}
-  {%- endif %}
-postfix_alias_database:
-  file.managed:
-    - name: {{ file_path }}
-    - source: salt://postfix/aliases
-    - user: root
-    - group: root
-    - mode: 644
-    - template: jinja
-    - require:
-      - pkg: postfix
-  {%- if need_newaliases %}
-  cmd.wait:
-    - name: newaliases
-    - cwd: /
-    - watch:
-      - file: {{ file_path }}
-  {%- endif %}
+{%- if 'present' in pillar.get('postfix', '')['aliases'] %}
```

Comment:
`pillar.get('postfix', '')` does not return a `dict` so `['aliases']` will just raise a rendering error.
You might just want to do jump right into the loop with `{%- for user, target in salt['pillar.get']('postfix:aliases:present', {}).items() %}`